runelite-client/src/main/java/net/runelite/client/plugins/microbot/tempoross/TemporossScript.java

```diff
@@ -0,0 +1,662 @@
+
+        if (!hasHarpoon()) {
+            // If we don't have a harpoon, set to the default harpoon that we can pick up
+            log("Missing selected harpoon, setting to default harpoon");
+            harpoonType = HarpoonType.HARPOON;
+            TemporossPlugin.setHarpoonType(harpoonType);
+
+            if (Rs2Player.isMoving() || Rs2Player.isInteracting())
+                return;
+
+            if (Rs2GameObject.interact(workArea.getHarpoonCrate(), "Take")) {
+                log("Taking harpoon");
+                sleepUntil(this::hasHarpoon, 10000);
+            }
+            return;
+        }
+
+        // Get bucket counts (empty and full)
+        int bucketCount = Rs2Inventory.count(item -> item.getId() == ItemID.BUCKET || item.getId() == ItemID.BUCKET_OF_WATER);
+        if ((bucketCount < temporossConfig.buckets() && state == State.INITIAL_CATCH) || bucketCount == 0) {
+            log("Buckets: " + bucketCount);
+            if (Rs2Player.isMoving() || Rs2Player.isInteracting())
+                return;
+
+            if (Rs2GameObject.interact(workArea.getBucketCrate(), "Take")) {
+                log("Taking buckets");
+                sleepUntil(() -> Rs2Inventory.waitForInventoryChanges(10000));
+            }
+            return;
+        }
+
+        int fullBucketCount = Rs2Inventory.count(ItemID.BUCKET_OF_WATER);
+        if (fullBucketCount <= 0) {
+            if (Rs2Player.isMoving() || Rs2Player.isInteracting())
+                return;
+
+            if (Rs2GameObject.interact(workArea.getPump(), "Use")) {
+                log("Filling buckets");
+                sleepUntil(() -> Rs2Inventory.count(ItemID.BUCKET) <= 0, 10000);
+            }
+            return;
+        }
+
+        if (temporossConfig.rope() && !temporossConfig.spiritAnglers() && !Rs2Inventory.contains(ItemID.ROPE)) {
+            if (Rs2Player.isMoving() || Rs2Player.isInteracting())
+                return;
+
+            if (Rs2GameObject.interact(workArea.getRopeCrate(), "Take")) {
+                log("Taking rope");
+                sleepUntil(() -> Rs2Inventory.waitForInventoryChanges(10000));
+            }
+            return;
+        }
+
+        if (temporossConfig.hammer() && !Rs2Inventory.contains(ItemID.HAMMER)) {
+            if (Rs2Player.isMoving() || Rs2Player.isInteracting())
+                return;
+
+            if (Rs2GameObject.interact(workArea.getHammerCrate(), "Take")) {
+                log("Taking hammer");
+                sleepUntil(() -> Rs2Inventory.waitForInventoryChanges(10000));
+            }
+        }
+    }
+    private void handleEnterMinigame() {
+        // Reset state variables
+        reset();
+
+        if (Rs2Player.isMoving() || Rs2Player.isAnimating()) {
+            return;
+        }
+        TileObject startingLadder = Rs2GameObject.findObjectById(ObjectID.ROPE_LADDER_41305);
+        if (startingLadder == null) {
+            return;
```

Comment:
Is it possible to add a Microbot.log here, to tell the user that the starting ladder is not found

Reply:
Added